src/conf_mode/containers.py

```diff
@@ -0,0 +1,218 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import os
+
+from vyos.config import Config
+from vyos.configdict import dict_merge
+from vyos.configdict import node_changed
+from vyos.configdict import leaf_node_changed
+from vyos import ConfigError
+from vyos.util import call, cmd, process_named_running
+from vyos.template import render
+from vyos import airbag
+import json
+airbag.enable()
+
+config_containers_registry = '/etc/containers/registries.conf'
+
+# Container management functions
+def container_exists(name):
+    '''
+       https://docs.podman.io/en/latest/_static/api.html#operation/ContainerExistsLibpod
+       Check if container exists. Response codes.
+       204 - container exists
+       404 - no such container
+    '''
+    c = cmd(f"curl --unix-socket /run/podman/podman.sock 'http://d/v3.0.0/libpod/containers/{name}/exists'")
+    # If container exists it return status code "0"
+    # This code not displayed
+    if c == "":
+        # Container exists
+        return True
+    else:
+        # Container not exists
+        return False
+
+def container_status(name):
+    '''
+       https://docs.podman.io/en/latest/_static/api.html#operation/ContainerInspectLibpod
+    '''
+    c = cmd(f"curl --unix-socket /run/podman/podman.sock 'http://d/v3.0.0/libpod/containers/{name}/json'")
+    data = json.loads(c)
+    status = data['State']['Status']
+
+    return status
+
+def container_stop(name):
+    c = cmd(f'podman stop {name}')
+
+def container_start(name):
+    c = cmd(f'podman start {name}')
+
+def ctnr_network_exists(name):
+    # Check explicit name for network.
+    c = cmd(f'podman network ls --quiet --filter name=^{name}$')
+    # If network name is found, return true
+    if bool(c) == True:
+        return True
+    else:
+        return False
+
+
+# Common functions
+def get_config(config=None):
+    if config:
+        conf = config
+    else:
+        conf = Config()
+    base = ['container']
+    container = conf.get_config_dict(base, get_first_key=True)
+
+    if 'name' in container or 'network' in container:
+        container['configured'] = True
+
+    # Delete container network, delete containers
+    dict = {}
+    tmp_net = node_changed(conf, ['container', 'network'])
+    if tmp_net:
+        dict = dict_merge({'net_remove' : tmp_net}, dict)
+        container.update(dict)
+
+    tmp_name = node_changed(conf, ['container', 'name'])
+    if tmp_name:
+        dict = dict_merge({'container_remove' : tmp_name}, dict)
+        container.update(dict)
+
+    # Registry ['docker.io'] should be set by default.
+    # Even if 'registry' doesn't set
+    if 'registry' not in container:
+        container['registry'] = []
+
+    container['registry'].append('docker.io')
```

Comment:
Please use CLI defaultValue option

Reply:
I am confused that this creates strange behavior problems, such as when a user enters a particular Registry, it is just code to actually pass two Registry.

1) such as when a user enters a particular Registry, it is just code to actually pass two Registry. So is it normal for Podman to pass both Registry?
2) Using DefaultValue might mean forcing the user to use a unique set value instead of a list
3) I don't know if you understand what I mean, but the problem now is that the current implementation causes strange configuration behavior in the behavior

I am not a core maintainer, but I request to modify the behavior and agree to @c-po 's request to avoid strange behavior as much as possible

Reply:
> Please use CLI defaultValue option

Done
```
vyos@r4-roll# set container name alp01 image alpine
[edit]
vyos@r4-roll# set container name alp01 allow-host-networks 
[edit]
vyos@r4-roll# commit
[ container ]
{'configured': True,
 'name': {'alp01': {'allow-host-networks': {}, 'image': 'alpine'}},
 'registry': ['docker.io']}

[edit]
vyos@r4-roll#
```